src/lib/libfs.js

```diff
@@ -199,7 +199,12 @@ FS.staticInit();
 
           if (parts[i] === '..') {
             current_path = PATH.dirname(current_path);
-            current = current.parent;
+            if (FS.isRoot(current)) {
+              path = current_path + '/' + parts.slice(i + 1).join('/');
```

Comment:
If the current node is the root node can we just assume current path is `/` here and do `path = '/' + parts.slice(i + 1).join('/')`

Reply:
No, because `FS.isRoot()` returns `true` if it's the root of a mount. So in the test, because we mounted NODEFS at `/working`, `FS.isRoot()` will return true when processing the `..` in the path `"/working/../other"`.